### ISSUE TITLE
Substance checker breaks on terminating comment

### ISSUE THREAD

**Issue by maxkrieger** [penrose/penrose#470]
In any domain, if your substance file ends in a comment:

```
Set A
Set B
Set C
Set D
-- Set E
```

You get:
```
 Cannot read property 'statements' of undefined
---
 * @param prog compiled AST of a Domain program
  150 |  * @param env  environment from the Domain checker
  151 |  */
> 152 | export const checkSubstance = (prog: SubProg, env: Env): CheckerResult => {
  153 |   const { statements } = prog;
  154 |   // check all statements
  155 |   const stmtsOk: CheckerResult = safeChain(statements, checkStmt, ok(env));
```

At least on `vanilla-renderer` branch -- havent tried other branches yet.

**Comment by wodeni** [comment]
Try pulling `main` again. Test passes without any changes to `main`: https://github.com/penrose/penrose/blob/fix-trailing-comment/packages/penrose-core/src/compiler/Substance.test.ts#L74

**Comment by wodeni** [comment]
Fixed in #471. Closing now.
